lib/ansible/plugins/lookup/aws_ssm.py

```diff
@@ -45,130 +48,140 @@
+                - aws_secret_key / aws_access_key / aws_security_token - AWS credentials
+                - region - the AWS region in which to do the lookup
+                - bypath - set to true to do a lookup of variables under a path
+                - recursive - set to true to do a recurse to paths below the path (needs bypath)
+            :return A list of parameter values or a list of dictionaries if doing a bypath lookup
         '''
 
-        ret = {}
-        response = {}
-        session = {}
-        ssm_dict = {}
-        lparams = {}
-
         if not HAS_BOTO3:
-            raise AnsibleError('botocore and boto3 are required.')
-
-        ssm_dict['WithDecryption'] = True
-
-        # check if option 'bypath' is specified, while still allowing to have a parameter with the same name
-        if 'bypath' in terms[1:]:
-            ssm_dict['Path'] = terms[0]
-            del terms[terms[1:].index('bypath') + 1]
-        else:
-            ssm_dict['Names'] = [terms[0]]
-
-        # Option to return short parameter names in by path lookups
-        if 'shortnames' in terms[1:]:
-            lparams['shortnames'] = True
-            del terms[terms[1:].index('shortnames') + 1]
-
-        if len(terms) > 1:
-            for param in terms[1:]:
-                if "=" in param:
-                    key, value = param.split('=')
-                else:
-                    raise AnsibleError("ssm parameter store plugin needs key=value pairs, but received %s" % param)
-
-                if key == "region" or key == "aws_profile":
-                    session[key] = value
-
-                # recurse or not
-                if key == "recursive" and boolean(value):
-                    ssm_dict['Recursive'] = True
-
-                # decrypt the value or not
-                if key == "decrypt" and boolean(value):
-                    ssm_dict['WithDecryption'] = False
-
-        if "aws_profile" in session:
-            boto3.setup_default_session(profile_name=session['aws_profile'])
+            raise AnsibleError('botocore and boto3 are required for aws_ssm lookup.')
 
-        if "region" in session:
-            client = boto3.client('ssm', region_name=session['region'])
-        else:
-            client = boto3.client('ssm')
+        ret = []
+        response = {}
+        ssm_dict = {}
 
-        try:
-            # Lookup by path
-            if 'Path' in ssm_dict:
-                response = client.get_parameters_by_path(**ssm_dict)
+        credentials = {}
+        credentials['boto_profile'] = boto_profile
+        credentials['aws_secret_access_key'] = aws_secret_key
+        credentials['aws_access_key_id'] = aws_access_key
+        credentials['aws_session_token'] = aws_security_token
+
+        client = _boto3_conn(region, credentials)
+
+        ssm_dict['WithDecryption'] = decrypt
+
+        # Lookup by path
+        if bypath:
+            for term in terms:
```

Comment:
paramlist needs to be instantiated before this line. Then after the loop len(terms) should be compared to the length of results. Also [line 172](https://github.com/ansible/ansible/pull/35569/files#diff-55e80dc4588cb031afcf2736c1338c47R172) should be removed or the 'return None' replaced with a continue